addOns/oast/src/main/java/org/zaproxy/addon/oast/services/boast/BoastService.java

```diff
@@ -47,6 +47,7 @@ public class BoastService extends OastService implements OptionsChangedListener
     private BoastParam boastParam;
     private int currentPollingFrequency;
     private ScheduledFuture<?> pollingSchedule;
+    private String latestPayload = null;
```

Comment:
Shouldn't be necessary.

Might need to change the check later to `isEmpty()`

Reply:
I changed the check to `if (latestPayload.isEmpty() || latestPayload == null) ` but it is misbehaving.